flake/deprecated.nix

```diff
@@ -1,19 +1,99 @@
+    {
+      ${output} = lib.mkIf (!lib.trivial.oldestSupportedReleaseIsAtLeast until) {
+        ${old} =
+          lib.warnIf (since != null -> lib.trivial.oldestSupportedReleaseIsAtLeast since)
+            "stylix: flake output `${output}.${system}.${lib.strings.escapeNixIdentifier old}` has been renamed to `${output}.${system}.${lib.strings.escapeNixIdentifier new}`."
+            config.${output}.${new};
+      };
+    };
+in
 {
-  # NOTE: the `flake` submodule has a `lazyAttrsOf` freeform type.
-  #
-  # This means a `mkIf false` definition will not omit the attr, because
-  # `lazyAttrsOf` adds an "empty value" stub throwing "used but not defined".
-  #
-  # Therefore, instead of doing `flake.foo = mkIf` you should do `flake = mkIf (…) { foo = …; }`.
-  # If you need multiple definitions with different conditions, use `flake = mkMerge [ … ]`.
+  options.stylix = {
+    perSystemAliases = lib.mkOption {
+      type = lib.types.listOf (
+        lib.types.submodule {
+          options = {
+            output = lib.mkOption {
+              type = lib.types.str;
+              description = "The per-system attribute in which to define the alias.";
+            };
+            old = lib.mkOption {
+              type = lib.types.str;
+              description = "The name of the alias.";
+            };
+            new = lib.mkOption {
+              type = lib.types.str;
+              description = "The name of the alias target.";
+            };
+            since = lib.mkOption {
+              type = with lib.types; nullOr ints.unsigned;
+              default = null;
+              description = ''
+                Warn only once the specified release is the oldest supported nixpkgs release.
+
+                If `null`, the alias will always warn.
+              '';
+            };
+            until = lib.mkOption {
+              type = lib.types.ints.unsigned;
+              description = ''
+                Create the alias only until the specified release is the oldest supported nixpkgs release.
+
+                The alias spec can be safely removed after this release.
+              '';
+            };
+          };
+        }
+      );
+      default = [ ];
+      description = "A list of per-system aliases.";
+    };
+  };
+
+  config = {
+    # NOTE: the `flake` submodule has a `lazyAttrsOf` freeform type.
+    #
+    # This means a `mkIf false` definition will not omit the attr, because
+    # `lazyAttrsOf` adds an "empty value" stub throwing "used but not defined".
+    #
+    # Therefore, instead of doing `flake.foo = mkIf` you should do `flake = mkIf (…) { foo = …; }`.
+    # If you need multiple definitions with different conditions, use `flake = mkMerge [ … ]`.
+
+    # Drop this alias after 26.05
+    flake = lib.mkIf (!lib.oldestSupportedReleaseIsAtLeast 2605) {
+      homeManagerModules = builtins.warn "stylix: flake output `homeManagerModules` has been renamed to `homeModules`" self.homeModules;
+    };
```

Comment:
Could this also be defined in `stylix.perSystemAliases` in a separate commit?

Reply:
modules are system agnostic, so they shouldn't be defined in a `perSystem` way.

We could add a separate `stylix.aliases` option, and maybe consider moving the current `stylix.perSystemAliases` option into the `perSystem` submodule as `perSystem.stylix.aliases`, which is where I probably should've put it TBH...

But given there's only one of each alias currently, I didn't want to invest too much time into coming up with a well designed solution :grin:

Reply:
> modules are system agnostic, so they shouldn't be defined in a `perSystem` way.
> 
> We could add a separate `stylix.aliases` option, and maybe consider moving the current `stylix.perSystemAliases` option into the `perSystem` submodule as `perSystem.stylix.aliases`, which is where I probably should've put it TBH...
> 
> But given there's only one of each alias currently, I didn't want to invest too much time into coming up with a well designed solution 😁

Sounds good. Let's leave it like this for now.